antarest/study/service.py

```diff
@@ -2375,3 +2425,40 @@ def get_disk_usage(self, uuid: str, params: RequestParameters) -> int:
         assert_permission(params.user, study, StudyPermissionType.READ)
         path = str(self.storage_service.get_storage(study).get_study_path(study))
         return get_disk_usage(path=path)
+
+    def get_matrix_with_index_and_header(
+        self, *, study_id: str, path: str, index: bool, header: bool, parameters: RequestParameters
+    ) -> pd.DataFrame:
+        matrix_path = Path(path)
+        study = self.get_study(study_id)
+        for aggregate in ["allocation", "correlation"]:
+            if matrix_path == Path("input") / "hydro" / aggregate:
+                all_areas = t.cast(
+                    t.List[AreaInfoDTO],
+                    self.get_all_areas(study_id, area_type=AreaType.AREA, ui=False, params=parameters),
+                )
+                if aggregate == "allocation":
+                    hydro_matrix = self.allocation_manager.get_allocation_matrix(study, all_areas)
+                else:
+                    hydro_matrix = CorrelationManager(self.storage_service).get_correlation_matrix(all_areas, study, [])  # type: ignore
+                return pd.DataFrame(data=hydro_matrix.data, columns=hydro_matrix.columns, index=hydro_matrix.index)
+
+        json_matrix = self.get(study_id, path, depth=3, formatted=True, params=parameters)
+        for key in ["data", "index", "columns"]:
+            if key not in json_matrix:
+                raise IncorrectPathError(f"The path filled does not correspond to a matrix : {path}")
+        if len(json_matrix["data"]) == 0:
+            return pd.DataFrame()
+        df_matrix = pd.DataFrame(data=json_matrix["data"], columns=json_matrix["columns"], index=json_matrix["index"])
+
+        if index:
+            matrix_index = self.get_input_matrix_startdate(study_id, path, parameters)
+            time_column = pd.date_range(
+                start=matrix_index.start_date, periods=len(df_matrix), freq=matrix_index.level.value[0]
+            )
+            df_matrix.index = time_column
+
+        for specific_matrix in SPECIFIC_MATRICES:
+            if re.compile(specific_matrix).match(path):
```

Comment:
I changed to `re.match(specific_matrix, path)` and it works for this path :
`input/hydro/prepro/de/energy` with the regex `input/hydro/prepro/*/energy` so i do not really understand what you meant by saying it would fail. 
Therefore, i keep the regex for now.